utilities/test_suite/HIP_NEW/Tensor_hip.cpp

```diff
@@ -0,0 +1,2406 @@
+            if (ip_bitDepth == 0)
+                rppt_color_to_greyscale_gpu(d_input, srcDescPtr, d_output, dstDescPtr, srcSubpixelLayout, handle);
+            else if (ip_bitDepth == 1)
+                rppt_color_to_greyscale_gpu(d_inputf16, srcDescPtr, d_outputf16, dstDescPtr, srcSubpixelLayout, handle);
+            else if (ip_bitDepth == 2)
+                rppt_color_to_greyscale_gpu(d_inputf32, srcDescPtr, d_outputf32, dstDescPtr, srcSubpixelLayout, handle);
+            else if (ip_bitDepth == 3)
+                missingFuncFlag = 1;
+            else if (ip_bitDepth == 4)
+                missingFuncFlag = 1;
+            else if (ip_bitDepth == 5)
+                rppt_color_to_greyscale_gpu(d_inputi8, srcDescPtr, d_outputi8, dstDescPtr, srcSubpixelLayout, handle);
+            else if (ip_bitDepth == 6)
+                missingFuncFlag = 1;
+            else
+                missingFuncFlag = 1;
+
+            break;
+        }
+        default:
+            missingFuncFlag = 1;
+            break;
+        }
+
+        hipDeviceSynchronize();
+        end = clock();
+        end_omp = omp_get_wtime();
+
+
+        if (missingFuncFlag == 1)
+        {
+            printf("\nThe functionality %s doesn't yet exist in RPP\n", func.c_str());
+            return -1;
+        }
+
+        gpu_time_used = ((double)(end - start)) / CLOCKS_PER_SEC;
+        omp_time_used = end_omp - start_omp;
+
+        if (gpu_time_used > max_time_used)
+            max_time_used = gpu_time_used;
+        if (gpu_time_used < min_time_used)
+            min_time_used = gpu_time_used;
+        avg_time_used += gpu_time_used;
+    }
+
+    if (test_type == 0)
+    {
+        gpu_time_used = gpu_time_used;
+        omp_time_used = omp_time_used;
+        cout << "\nGPU Time - Tensor : " << gpu_time_used;
+        cout << "\nOMP Time - Tensor : " << omp_time_used;
+        printf("\n");
+    }
+    else
+    {
+        avg_time_used /= num_iterations;
+        max_time_used = max_time_used;
+        min_time_used = min_time_used;
+        avg_time_used = avg_time_used;
+        // Display measured times
+
+        cout << fixed << "\nmax,min,avg in ms = " << max_time_used << "," << min_time_used << "," << avg_time_used << endl;
+    }
+
+    if (test_type == 0)
+    {
+        // Reconvert other bit depths to 8u for output display purposes
+        string fileName = std::to_string(ip_bitDepth);
+        ofstream outputFile(fileName + ".csv");
+        ofstream outFile(func + ".csv");
+
+        if (ip_bitDepth == 0)
+        {
+            hipMemcpy(output, d_output, oBufferSizeInBytes_u8, hipMemcpyDeviceToHost);
+            Rpp8u *outputTemp;
+            outputTemp = output + dstDescPtr->offsetInBytes;
+
+            if (outputFile.is_open())
+            {
+                for (int i = 0; i < oBufferSize; i++)
+                {
+                    outputFile << (Rpp32u) *outputTemp << ",";
+                    outFile << (Rpp32u) *outputTemp << ",";
```

Comment:
Why do we write two times to outFile and outputFile?

Reply:
Done